ts/WoltLabSuite/Core/Ui/ItemList/LineBreakSeparatedText.ts

```diff
@@ -0,0 +1,258 @@
+      confirm: () => {
+        button.closest("li")!.remove();
+
+        if (this.itemList.childElementCount === 0) {
+          this.hideList();
+        }
+
+        this.items.delete(item);
+      },
+      message: Language.get("wcf.button.delete.confirmMessage", {
+        objectTitle: item,
+      }),
+      messageIsHtml: true,
+    });
+  }
+
+  /**
+   * Hides the item list and clear button.
+   */
+  protected hideList(): void {
+    DomUtil.hide(this.itemList);
+    DomUtil.hide(this.clearButton!);
+  }
+
+  /**
+   * Adds the initial values to the list.
+   */
+  protected initValues(): void {
+    Array.from(this.itemList.children).forEach((el: HTMLElement) => {
+      this.items.add(el.dataset.value!);
+
+      el.querySelector(".jsDeleteItem")!.addEventListener("click", (ev) => this.deleteItem(ev));
+    });
+  }
+
+  /**
+   * Inserts the given item to the list.
+   */
+  protected insertItem(item: string): void {
+    this.items.add(item);
+
+    const itemElement = document.createElement("li");
+    itemElement.dataset.value = item;
+
+    const deleteButton = document.createElement("span");
+    deleteButton.classList.add("icon", "icon16", "fa-times", "jsDeleteItem", "jsTooltip", "pointer");
+    deleteButton.title = Language.get("wcf.global.button.delete");
+    deleteButton.addEventListener("click", (ev) => this.deleteItem(ev));
+    itemElement.append(deleteButton);
+
+    itemElement.append(document.createTextNode(" "));
+
+    const label = document.createElement("span");
+    label.innerText = item;
+    itemElement.append(label);
+
+    Array.from(this.itemList.children).some((el: HTMLElement) => {
+      if (el.dataset.value!.localeCompare(item) === 1) {
+        this.itemList.insertBefore(itemElement, el);
+
+        return true;
+      }
+    });
+
+    if (!itemElement.parentElement) {
+      this.itemList.append(itemElement);
+    }
+
+    this.showList();
+  }
+
+  /**
+   * Adds an item to the list when pressing "Enter" in the input field.
+   */
+  protected keydown(event: KeyboardEvent): void {
+    if (event.key === "Enter") {
+      this.addItem(event);
+    }
+  }
+
+  /**
+   * Adds multiple items at one to the list when pasting multiple lines of text into the input
+   * field.
+   */
+  protected paste(event: ClipboardEvent): void {
+    const items = event.clipboardData!.getData("text/plain").split("\n");
+    if (items.length > 1) {
+      event.preventDefault();
+
+      items.forEach((item) => this.insertItem(item));
+
+      this.resetInput();
+    }
+  }
+
+  /**
+   * Resets the input field.
+   */
+  protected resetInput(): void {
+    DomUtil.innerError(this.itemInput!.parentElement!, "");
+    this.itemInput!.value = "";
+  }
+
+  /**
+   * Shows the item list and clear button.
+   */
+  protected showList(): void {
+    DomUtil.show(this.itemList);
+    DomUtil.show(this.clearButton!);
+  }
+
+  /**
+   * Adds a hidden input field with the data to the form before it is submitted.
+   */
```

Comment:
I prefer a continuous sync of the fields (unless it comes with a huge performance hit), because it cooperates better with additional submit handlers for the same form. (see also my remark in the template).

Reply:
This behavior is consistent with `Ui/ItemList`.